### PR TITLE
[WIP] refactor and optimize iceberg

### PR DESCRIPTION
**Pull request by stephen-shelby**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/ExpressionContext.java [hunk 1]
```diff
@@ -118,6 +118,7 @@ public Statistics getStatistics() {
         return statistics;
     }
 
+
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/ExpressionContext.java:121]: extra line

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergConnector.java [hunk 2]
```diff
@@ -15,50 +15,92 @@
...
     private final String catalogName;
-    private ConnectorMetadata metadata;
+    private final IcebergCatalog icebergNativeCatalog;
 
     public IcebergConnector(ConnectorContext context) {
         this.catalogName = context.getCatalogName();
         this.properties = context.getProperties();
         this.cloudConfiguration = CloudConfigurationFactory.tryBuildForStorage(properties);
-        this.hdfsEnvironment = new HdfsEnvironment(null, cloudConfiguration);
+        this.hdfsEnvironment = new HdfsEnvironment(properties, cloudConfiguration);
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergConnector.java:53]: You can pass as `new HdfsEnvironment(null, cloudConfiguration)`

File: fe/fe-core/src/test/java/com/starrocks/connector/iceberg/IcebergMetadataTest.java [hunk 3]
```diff
@@ -56,7 +57,7 @@ public void testListDatabaseNames(@Mocked HiveMetaStoreThriftClient metaStoreThr
         properties.put(ICEBERG_METASTORE_URIS, metastoreUris);
         properties.put(ICEBERG_CATALOG_TYPE, "hive");
         HdfsEnvironment hdfsEnvironment = new HdfsEnvironment();
-        IcebergMetadata metadata = new IcebergMetadata(CATALOG_NAME, properties, hdfsEnvironment);
```
Comment (Smith-Cruise) [inline fe/fe-core/src/test/java/com/starrocks/connector/iceberg/IcebergMetadataTest.java:59]: Looks like `CATALOG_NAME` and `hdfsEnvironment` not used anymore, remove it.

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Utils.java [hunk 4]
```diff
@@ -366,21 +361,7 @@ public static boolean hasUnknownColumnsStats(OptExpression root) {
...
-                    Expression icebergPredicate = new ScalarOperatorToIcebergExpr().convert(predicates, icebergContext);
-                    List<ColumnStatistic> columnStatisticList = IcebergTableStatisticCalculator.getColumnStatistics(
-                            icebergPredicate, table.getIcebergTable(),
-                            scanOperator.getColRefToColumnMetaMap());
-                    return columnStatisticList.stream().anyMatch(ColumnStatistic::isUnknown);
-                } catch (Exception e) {
-                    LOG.warn("Iceberg table {} get column failed. error : {}", table.getName(), e);
-                    return true;
-                }
+                return true;
```
[the remark this answers is not in the document]
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Utils.java:364]: iceberg metadata doesn't has ndv

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Comment by stephen-shelby** [comment]
ref https://github.com/StarRocks/starrocks/issues/18560

**Pull request closed without merging**